lib/onelogin/ruby-saml/settings.rb

```diff
@@ -377,6 +382,12 @@ def get_sp_certs_multi
         certs.each { |_, ary| ary.freeze }
         certs
       end
+
+      def build_cert_object(cert)
+        return cert if cert.is_a?(OpenSSL::X509::Certificate)
+
+        OneLogin::RubySaml::Utils.build_cert_object(cert)
```

Comment:
Why not adapt directly the Utils.build_cert_object to be able to handle string and OpenSSL::X509::Certificate?

Reply:
Yes, I was thinking about that, too. But then I decided to catch the problem as early as possible. And if it's fixed in Utils, it creates kind of a dependency. Utils would fix problems that come from Settings. Settings would not correctly work on it's own.